src/Surfnet/StepupGateway/SecondFactorOnlyBundle/Tests/Service/Gateway/GatewayLoginServiceTest.php

```diff
@@ -0,0 +1,378 @@
+<?php
+
+/**
+ * Copyright 2017 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupGateway\SecondFactorOnlyBundle\Test\Service\Gateway;
+
+use DateTime;
+use Mockery;
+use SAML2\Response;
+use Surfnet\SamlBundle\Http\HttpBindingFactory;
+use Surfnet\SamlBundle\Http\RedirectBinding;
+use Surfnet\SamlBundle\Entity\IdentityProvider;
+use Surfnet\SamlBundle\Entity\ServiceProvider;
+use Surfnet\SamlBundle\Http\PostBinding;
+use Surfnet\SamlBundle\Monolog\SamlAuthenticationLogger;
+use Surfnet\SamlBundle\SAML2\ReceivedAuthnRequest;
+use Surfnet\StepupBundle\Service\LoaResolutionService;
+use Surfnet\StepupGateway\SamlStepupProviderBundle\Saml\ProxyResponseFactory;
+use Surfnet\StepupGateway\SecondFactorOnlyBundle\Saml\ResponseFactory;
+use Surfnet\StepupGateway\SecondFactorOnlyBundle\Service\LoaResolutionService as SecondFactorLoaResolutionService;
+use Surfnet\StepupBundle\Service\SecondFactorTypeService;
+use Surfnet\StepupBundle\Value\Loa;
+use Surfnet\StepupGateway\GatewayBundle\Entity\SecondFactor;
+use Surfnet\StepupGateway\GatewayBundle\Saml\AssertionSigningService;
+use Surfnet\StepupGateway\GatewayBundle\Saml\Proxy\ProxyStateHandler;
+use Surfnet\StepupGateway\GatewayBundle\Saml\ResponseContext;
+use Surfnet\StepupGateway\SecondFactorOnlyBundle\Service\Gateway\SecondFactorLoginService;
+use Surfnet\StepupGateway\SecondFactorOnlyBundle\Service\Gateway\SecondFactorRespondService;
+use Surfnet\StepupGateway\GatewayBundle\Service\SamlEntityService;
+use Surfnet\StepupGateway\GatewayBundle\Service\SecondFactorService;
+use Surfnet\StepupGateway\GatewayBundle\Tests\TestCase\GatewaySamlTestCase;
+use Surfnet\StepupGateway\SecondFactorOnlyBundle\Service\LoaAliasLookupService;
+use Surfnet\StepupGateway\SecondFactorOnlyBundle\Service\SecondFactorOnlyNameIdValidationService;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\HttpFoundation\Session\Storage\MockArraySessionStorage;
+
+/**
+ * Class GatewayLoginServiceTest
+ * @package Surfnet\StepupGateway\GatewayBundle\Tests\Service
+ *
+ * TODO: test handle binding
+ *
+ */
+final class GatewayLoginServiceTest extends GatewaySamlTestCase
```

Comment:
The name of this test is the same as the one found in the gateway bundle. Copy pasta? I'd expect something like SecondFactorOnlySsoServiceTest

Reply:
Changed